lib/utils/credentials.js

```diff
@@ -1,7 +1,16 @@
 import { IDENTITY_DERIVATION_ROUNDS } from './constants'
 
-export async function deriveNoteCreds(idConfig, crypto, username, password) {
-  const noteID = await crypto.genericHash(username + idConfig.realmId)
+export async function deriveNoteCreds(
+  idConfig,
+  crypto,
+  username,
+  password,
+  broker = false
+) {
+  const nameSeed = broker
+    ? `broker${username}${idConfig.realmId}`
```

Comment:
This was the name above `broker${username}${this.config.realmId}key` but now its missing `key` did I miss a note?

Reply:
There are three notes:

- Login note
- Broker Key Note
- Broker Login Note

Each needs a different name, this helper gets the ID fro the two login notes, while the above 'key' note is the note containing the broker key. It gets a little muddy, so it's probably at least worth commenting to help clear it up a bit.